include/range/v3/action/remove.hpp

```diff
@@ -0,0 +1,82 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Andrey Diduh 2019
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_ACTION_REMOVE_HPP
+#define RANGES_V3_ACTION_REMOVE_HPP
+
+#include <meta/meta.hpp>
+#include <range/v3/core.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/action/action.hpp>
+#include <range/v3/action/remove_if.hpp>
+
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-actions
+        /// @{
+        namespace action
+        {
+            struct remove_fn
+            {
+            private:
+                friend action_access;
+                template<typename V, typename P = ident, CONCEPT_REQUIRES_(!Range<V>())>
+                static auto bind(remove_fn remove, const V& value, P proj = P{})
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                    std::bind(remove,
+                        std::placeholders::_1,
+                        std::cref(value),
+                        protect(std::move(proj))
+                    )
+                )
+
+                template<class Value>
```

Comment:
range-v3 refuses to accept the self-evident truth that `class` is superior to `typename` for template type parameters:
```suggestion
                template<typename Value>
```

Reply:
I find that C++'s addition of a `class` keyword in the 80's was largely pointless and a cowing to the OO religion of the day. `struct` was doing just fine and didn't need replacement. In range-v3 I voted with my feet (er, fingers) and type `c-l-a-s-s` hardly ever.